compiler/rustc_ast_passes/src/feature_gate.rs

```diff
@@ -701,11 +701,16 @@ pub fn check_crate(krate: &ast::Crate, sess: &Session) {
     visit::walk_crate(&mut visitor, krate);
 }
 
-fn maybe_stage_features(sess: &Session, krate: &ast::Crate) {
+fn maybe_stage_lang_features(sess: &Session, krate: &ast::Crate) {
+    // checks if `#![feature]` has been used to enable any lang feature
+    // does not check the same for lib features
```

Comment:
Hmm, this isn't always true though, right? It will still error on lib features if there's at least one lang feature.

Reply:
but the check is only on lang_features. It will error only because of that one lang feature

Reply:
I don't think that's true. The check is `for attr in krate.attrs.iter().filter(|attr| attr.has_name(sym::feature)) {`, which doesn't distinguish the two.

Please write a test, it will be much easier than trying to figure it out from the code.

Reply:
Ok yup, https://github.com/rust-lang/rust/pull/89012#issuecomment-921068622 is what I expected. Please update the comment and then I think this is good to merge :)

Reply:
and I think it makes sense to revert changing the name of the function, too - it still checks lib features when there's only one library feature.